encrypted-media/scripts/playback-retrieve-persistent-usage-record.js

```diff
@@ -0,0 +1,121 @@
+            assert_equals(event.target, _video);
+            assert_true(event instanceof window.MediaEncryptedEvent);
+            assert_equals(event.type, 'encrypted');
+
+            waitForEventAndRunStep('message', _mediaKeySession, onMessage, test);
+            _mediaKeySession.generateRequest(   config.initDataType || event.initDataType,
+                                                config.initData || event.initData ).then( function() {
+
+                _sessionId = _mediaKeySession.sessionId;
+            }).catch(function(error) {
+                forceTestFailureFromPromise(test, error);
+            });
+        }
+        
+        function onMessage(event) {
+            assert_equals( event.target, _mediaKeySession );
+            assert_true( event instanceof window.MediaKeyMessageEvent );
+            assert_equals( event.type, 'message');
+
+            assert_in_array(  event.messageType, [ 'license-request', 'individualization-request' ] );
+
+            config.messagehandler( config.keysystem, event.messageType, event.message ).then( function( response ) {
+
+                _mediaKeySession.update( response )
+                .catch(function(error) {
+                    forceTestFailureFromPromise(test, error);
+                });
+            });
+        }
+        
+        function onPlaying(event) {
+
+            // Not using waitForEventAndRunStep() to avoid too many
+            // EVENT(onTimeUpdate) logs.
+            _video.addEventListener('timeupdate', onTimeupdate, true);
+        }
+
+        function onTimeupdate(event) {
+            if ( _video.currentTime > ( config.duration || 5 ) ) {
+
+                _video.removeEventListener('timeupdate', onTimeupdate );
+
+                _video.pause();
+
+                _mediaKeySession.closed.then( test.step_func( onClosed ) );
+               
+                _mediaKeySession.close();
+            }
+        }
+        
+        function onClosed(event) {
+
+            _video.src = "";
+            _video.setMediaKeys( null );
+            
+            var win = window.open( config.windowscript );
+            window.addEventListener('message', test.step_func(function( event ) {
+                
+                event.data.forEach(test.step_func(function( assertion ) {
+                
+                    assert_equals(assertion.actual, assertion.expected, assertion.message);
+                    
+                }));
+                
+                win.close();
+                
+                test.done();
+            }));
+            
+            delete config.video;
+            delete config.messagehandler;
+            
+            win.onload = function() {
+                
+                win.postMessage( { config: config, sessionId: _sessionId }, '*' );
+            }
+
+        }
+
+        navigator.requestMediaKeySystemAccess(config.keysystem, [ configuration ]).then(function(access) {
```

Comment:
This test case checks that the record can be retrieved later in another window. In onClosed, we open another window and post the config and sessionId to it. The other window sends back a set of assertions to be checked.

The page to be loaded in the other window is supplied in the config and lives in the resources folder.